### PR TITLE
auth: Adding ALLOWED_USERS config.

### PR DESCRIPTION
**Pull request by amerine**
This lets you add a semicolon-delimited list of additional email
addresses that would be considered to be valid users.

cc/ @voidlock @freeformz @finack

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -65,6 +66,16 @@ func authorize(s sessions.Session, rw http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func checkAllowed(e string) bool {
```
Comment (freeformz) [inline main.go]: May as well move both checks into this function.

**Comment by amerine** [comment]
@freeformz Good idea! Done.

**Comment by freeformz** [comment]
:+1: with 2 notes:

1. I think sproxy should ask the splunk API directly if a user exists by email address and if that email address exists allows them in. This way we'd only have to create the accounts and can remove the whitelist stuff entirely
1. this uses google's openid stuff. a login to salesforce relies on Aloha. I have no idea how those interact with openid, so this approach may not work for salesforce anyway.

**Comment by da3da1us** [comment]
@amerine So if I'm reading this right, anything can be added to the list of allowed users? I was imagining keeping it restricted to SFDC addresses. Also how would this whitelist be maintained?

**Comment by amerine** [comment]
@da3da1us You're correct, anything can be added. I didn't make it SFDC specific because sproxy is OSS. 

The whitelist would be maintained via our normal Heroku config vars.

**Comment by finack** [comment]
@da3da1us this would only whitelist what domains are allowed to be proxied. Splunk itself has the canonical list of which accounts are actually allowed and with what role.

**Comment by voidlock** [comment]
I'm still not convinced that this would work because of the comment from @freeformz 

> this uses google's openid stuff. a login to salesforce relies on Aloha. I have no idea how those interact with openid, so this approach may not work for salesforce anyway.

**Comment by freeformz** [comment]
I did not mean to cause FUD.
Merge it, try it and if it works great! If not then we can investigate
further.

**Comment by amerine** [comment]
This is stale. Closing.

**Pull request closed without merging**